### PR TITLE
Switch to log.info for deterministic mode

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

Switch to `log.info` as this isn't something going wrong that requires user intervention. If a user intentionally sets this, it should not result in a warning.

Most frequently, this results in spurious warnings in our Ci/CD

### CODE REVIEW

File: composer/utils/reproducibility.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def configure_deterministic_mode():
     # See https://pytorch.org/docs/stable/generated/torch.use_deterministic_algorithms.html
     # and https://docs.nvidia.com/cuda/cublas/index.html#cublasApi_reproducibility
     os.environ['CUBLAS_WORKSPACE_CONFIG'] = ':4096:8'
-    warnings.warn('Deterministic mode is activated. This will negatively impact performance.', category=UserWarning)
+    log.warning('Deterministic mode is activated. This will negatively impact performance.', category=UserWarning)
```
Comment (snarayan21) [inline composer/utils/reproducibility.py:130]: log.info?

**Review by snarayan21 — COMMENTED** [review]
according to PR description, likely log.info right?

File: composer/utils/reproducibility.py (hunk 1 above)
Comment (mvpatel2000) [reply composer/utils/reproducibility.py]: Oops forgot to repush

**Review by snarayan21 — APPROVED** [review]
lgtm

**Pull request merged**